index.d.ts

```diff
@@ -10,226 +10,233 @@ declare global {
-        publicKey?: string,
-        options?: SignMessageOptions
-      ): Promise<boolean>;
-
-      /**
-       * Experimental event emitter. Allows listening to gateway config
-       * updates, bundler node changes, etc.
-       */
-      events: Emitter<InjectedEvents>;
-    };
+    arweaveWallet: ArweaveWallet;
   }
+  
   interface WindowEventMap {
     walletSwitch: CustomEvent<{ address: string }>;
     arweaveWalletLoaded: CustomEvent<{}>;
   }
 }
 
+export type TypedArray =
+  | Int8Array
+  | Uint8Array
+  | Uint8ClampedArray
+  | Int16Array
+  | Uint16Array
+  | Int32Array
+  | Uint32Array
+  // | Float16Array
+  | Float32Array
+  | Float64Array
+  | BigInt64Array
+  | BigUint64Array;
+
+export type BinaryDataType = ArrayBuffer | TypedArray | DataView; // | Buffer;
+
+export interface ArweaveWallet {
+  /**
+   * Name of the wallet the API was provided by.
+   */
+  walletName: string;
+
+  /**
+   * Semver type version of the wallet
+   */
+  walletVersion: string;
+
+  /**
+   * Connect to ArConnect and request permissions. This function can always be
+   * called again if you want to request more permissions for your site.
+   *
+   * @param permissions
+   * @param appInfo
+   */
+  connect(
+    permissions: PermissionType[],
+    appInfo?: AppInfo,
+    gateway?: GatewayConfig
+  ): Promise<void>;
+
+  /**
+   * Disconnect from ArConnect. Removes all permissions from your site.
+   */
+  disconnect(): Promise<void>;
+
+  /**
+   * Get the currently used wallet's address in the extension.
+   *
+   * @returns Promise of wallet address string
+   */
+  getActiveAddress(): Promise<string>;
+
+  /**
+   * Get all addresses added to the ArConnect extension
+   *
+   * @returns Promise of a list of the added wallets' addresses.
+   */
+  getAllAddresses(): Promise<string[]>;
+
+  /**
+   * Get wallet names for addresses.
+   *
+   * @returns Promise of an object with addresses and wallet names
+   */
+  getWalletNames(): Promise<{ [addr: string]: string }>;
+
+  /**
+   * Sign a transaction.
+   *
+   * @param transaction A valid Arweave transaction without a wallet keyfile added to it
+   * @param options Arweave signing options
+   *
+   * @returns Promise of a signed transaction instance
+   */
+  sign(
+    transaction: Transaction,
+    options?: SignatureOptions
+  ): Promise<Transaction>;
+
+  /**
+   * Get the permissions allowed for you site by the user.
+   *
+   * @returns Promise of a list of permissions allowed for your dApp.
+   */
+  getPermissions(): Promise<PermissionType[]>;
+
+  /**
+   * Encrypt a string, using the user's wallet.
+   *
+   * @param data Buffer-like structure to encrypt.
+   * @param options Encrypt options. See https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/encrypt#algorithm.
+   *
+   * @returns Promise of the encrypted `Uint8Array`.
+   */
+  encrypt(
+    data: BinaryDataType,
+    options?: RsaOaepParams | AesCtrParams | AesCbcParams | AesGcmParams,
```

Comment:
Before:

```
        options: {
          algorithm: string;
          hash: string;
          salt?: string;
        }
```

Which didn't match the docs:

![image](https://github.com/user-attachments/assets/312f404b-2aae-44f5-81cf-4ce0af906d45)